src/vs/vscode.d.ts

```diff
@@ -6145,6 +6160,31 @@ declare module 'vscode' {
 		 * @returns a [TreeView](#TreeView).
 		 */
 		export function createTreeView<T>(viewId: string, options: { treeDataProvider: TreeDataProvider<T> }): TreeView<T>;
+
+		/**
+		 * Registers a [Uri handler](#UriHandler) capable of handling system-wide [Uris](#Uri).
+		 * In case there are multiple windows open, the topmost window will handle the Uri.
+		 * A Uri handler is scoped to the extension it is contributed from; it will only
+		 * be able to handle Uris which are directed to the extension itself. The Uri format
+		 * is predetermined by the extension's identifier.
+		 *
+		 * For example, if the `vscode.git` extension registers a Uri handler, it will only
+		 * be allowed to handle Uris with the prefix `{scheme}://vscode.git`, in which `{scheme}`
+		 * is either `vscode` or `vscode-insiders`. All the following Uris are examples:
```

Comment:
This is tricky, we don't use product-names in the documentation. I usually use 'the editor' but something better is needed for those samples. 

Also, add some clarity or familiar uri-terms, like: The uri-scheme must be the product name, the uri-authority must be the extension id, the path, query, and fragment can be chosen freely.